src/sherpa_ai/agents/qa_agent.py

```diff
@@ -96,6 +100,32 @@ def synthesize_output(self) -> str:
             self.belief.get_internal_history(self.llm.get_num_tokens),
         )
 
+        count = 0
+        while count < self.validation_count:
```

Comment:
Can you factor out this number validation process into a new function? We may want to put this feature available for other agents as well in the future.